### PR TITLE
Fix: Reset cache on first message request.

### LINKED ISSUE #773: [Android] Anchor position not maintained on first message list load
The anchor position is not maintained whenever the message list is loaded for the first time.

### PR DESCRIPTION
**Pull request by Sam1301**
fixes #773 

This solves the anchor bug on both platforms.

### CODE REVIEW

File: src/app/appReducers.js [hunk 1]
```diff
@@ -38,12 +39,20 @@ export default (state: AppState = initialState, action: Action) => {
         ...state,
         needsInitialFetch: !!action.apiKey,
       };
-    case REHYDRATE:
+    case REHYDRATE: {
+      const needsInitialFetch = !!getAuth(action.payload).apiKey;
+      if (action.payload.chat && needsInitialFetch) {
+        // note: Reassigning a variable reference is not mutation
```
Comment (borisyankov) [inline src/app/appReducers.js]: Thanks for the link, so I can see you are wrong 😄 
Reassigning a variable like
```js
state = undefined;
```
does not mutate anything, that is true.

Doing this though:
```js
state.something = undefined;
```
You are mutating `state`, and you are not mutating `something`.

Comment (borisyankov) [inline src/app/appReducers.js]: You can try this with `deepFreeze()`.

**Comment by Sam1301** [comment]
Updated the branch 👍

**Comment by borisyankov** [comment]
Thanks 👍

**Pull request merged**